### ISSUE TITLE
vscode.startDebug with type "node" vs. "node2"

### ISSUE THREAD

**Issue by SAKryukov** [Microsoft/vscode#30865]
"vscode.startDebug" does not respect `type: "node2"` when the configuration supplies `type: "node"`, as it happens in the embedded extension for node.js debugging.

## Background

Visual Studio Code tracing shows that the embedded extension for node.js debugging actually uses `type: "node2"`, even when the auto-generated file "launch.json" provides `type: "node"`. Moreover, `type: "node"` if supplied to the "complex command" "vscode.startDebug" does not work anymore (see steps to reproduce below).

This is very confusing and not documented. The cases of "node2" and "node" should be merged to eliminate "node2". In worst case, this peculiarity should be thoroughly documented.

Related to [24762](https://github.com/Microsoft/vscode/issues/24762).

### 1) Try node.js debugging with "launch.json":

```
{
    "version": "0.1.0",
    "configurations": [
        {
            "name": "Launch Extension",
            "type": "extensionHost",
            "request": "launch",
            "program": "${file}",
            "protocol": "auto" // anything but "legacy"
            ],
        }
    ]
}
```
It does work.

### 2) Now, in some extension code:

```
        const launchConfiguration = {
            type: "node", // a real confusion! found by tracing Visual Studio Code
            name: "Launch Extension",
            request: "launch",
            stopOnEntry: false,
            protocol: "auto"
        };
        vscode.commands.executeCommand("vscode.startDebug", launchConfiguration);
```
It shows the error:
> [DEP0062] DeprecationWarning: `node --debug` and `node --debug-brk` are invalid. Please use `node --inspect` or `node --inspect-brk` instead

### 3) The workaround:

Change `type: "node"` to `type: "node2"`. It works.

## 
Reproduces without extensions: No (the problem appears in extension, more exactly)

**Comment by weinand** [comment]
The debug type "node" is the official type for node debugging. The internal debug type "node2" is an implementation detail of node-debug. It was introduced because node.js v8.x required a new debug protocol and we could not provide a single implementation for the old ("legacy") and the new ("inspector") protocol. The extension code in node-debug uses various kinds of probing to figure out which protocol to use and switches between the two protocol implementations via the debug types "node" and "node2". The public face of this is the "protocol" attribute and its two values "legacy" and "inspector".

Since "node2" is a temporary implementation details it can disappear at any time and there is neither a need nor an intention to document it.

I will close this issue but recommend that you create an new issue that explains what you trying to achieve without referring to the implementation detail "node2". Just use "inspector protocol" or "both protocols" instead... ;-)

**Comment by SAKryukov** [comment]
You would be perfectly right if it wasn't simply a bug.

I already mentioned this "trying to achieve", perhaps not clearly enough — I mentioned
```
 "protocol": "auto" // anything but "legacy"
```
I mean, I also tried ```protocol: "inspector"```. Of course I did. Moreover, by documentation, *it should* work even with `"node"` and "`auto`", because the system can see what version of node is used and "auto" apply right protocol. This is what actually happens.

I guess, I'll have to open new issue and mention "protocol": "inspector" case explicitly. I hope you test it instead of closing the issue.

Thank you.
—SA
